src/java.base/share/classes/java/lang/invoke/IndirectVarHandle.java

```diff
@@ -95,14 +101,15 @@ VarHandle target() {
     MethodHandle getMethodHandle(int mode) {
         MethodHandle handle = handleMap[mode];
         if (handle == null) {
-            MethodHandle targetHandle = target.getMethodHandle(mode); // might throw UOE of access mode is not supported, which is ok
-            handle = handleMap[mode] = handleFactory.apply(AccessMode.values()[mode], targetHandle);
+            /* OpenJ9 pre-initializes the handleMap in the constructor. So, there is no need to reapply handleFactory here. */
+            handle = target.getMethodHandle(mode); // might throw UOE of access mode is not supported, which is ok
         }
         return handle;
     }
 
     @Override
     public MethodHandle toMethodHandle(AccessMode accessMode) {
-        return getMethodHandle(accessMode.ordinal()).bindTo(directTarget);
+        MethodHandle mh = getMethodHandle(accessMode.ordinal());
+        return MethodHandles.insertArguments(mh, mh.type().parameterCount() - 1, directTarget);
```

Comment:
OpenJ9 inserts the VarHandle argument at the end whereas the reference implementation inserts the VarHandle argument at the beginning.